### PR TITLE
more errors for the people

### PR DESCRIPTION
**Pull request by yaleman**
# Change summary

- more explainable errors coming back from things in unix-land because I'm getting errors but ... everything seems to be working? 😄 

Checklist

- [x] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: unix_integration/common/src/unix_proto.rs [hunk 1]
```diff
@@ -177,7 +177,7 @@ pub enum ClientResponse {
     ProviderStatus(Vec<ProviderStatus>),
 
     Ok,
-    Error,
+    Error(String),
```
Comment (Firstyear) [inline unix_integration/common/src/unix_proto.rs:180]: I'm not a fan of String in error types. I think it was a mistake that we did it in the past, so we should probably add stronger types here I think.

Comment (yaleman) [reply unix_integration/common/src/unix_proto.rs:180]: So swap to an enum?

Comment (yaleman) [reply unix_integration/common/src/unix_proto.rs:180]: And in that case, should I be using the OperationError enum as the one true error place? 😄

Comment (yaleman) [reply unix_integration/common/src/unix_proto.rs]: (done)

**Review by Firstyear — APPROVED** [review]

**Pull request merged**